### PR TITLE
Cleanup widget and fix memory leaks

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR fixes a memory leak in the widget react library (the widget itself was fine)

There was no proper cleanup being done.

## Test

Reproduce the issue this PR tries to solve:
https://github.com/cowprotocol/cowswap/pull/4263

Test the same steps in https://github.com/cowprotocol/cowswap/pull/4263 but in this PR:
- https://widget-configurator-git-widget-fix-performance-issue-cowswap.vercel.app/

If all goes well, it should not freeze ♨️😎

### CODE REVIEW

File: apps/widget-configurator/src/app/configurator/index.tsx
Comment (anxolin) [inline apps/widget-configurator/src/app/configurator/index.tsx:201]: this will be reverted after I get some approvals in this PR

File: libs/widget-react/src/lib/CowSwapWidget.tsx
Comment (anxolin) [inline libs/widget-react/src/lib/CowSwapWidget.tsx:38]: Actual fix

**Review by elena-zh — APPROVED** [review]
Great, thank you!

**Review by alfetopito — APPROVED** [review]

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**
